### PR TITLE
[Survival] adding the 5 spec specific legendaries 

### PR DESCRIPTION
**Pull request by Pewtro**
This adds the 5 spec specific legendaries. 

Helbrine: 
image
Report: https://www.warcraftlogs.com/reports/QtDF2PBh4Aa1Xxgr/#fight=22&source=11

Frizzo's: 
image
Report: https://www.warcraftlogs.com/reports/AbdZzH3WFxGhyYfN#fight=9&type=damage-done&source=16

Unseen Predator's Cloak: 
image
Report: https://www.warcraftlogs.com/reports/ZxVMjQCLfqcRmtKH#fight=8&type=auras&translate=true&hostility=1&spells=debuffs&target=19

Butcher's Bone Apron: 
image
Report: https://www.warcraftlogs.com/reports/Rt9cXWGzC7bhdL6v#fight=1&type=damage-done

Nesingwary's Trapping Treads:
image

Report: https://www.warcraftlogs.com/reports/nWMF8h1vTq3cbCKw#fight=1&type=damage-done

### CODE REVIEW

File: src/Parser/Hunter/Survival/Modules/Items/NesingwarysTrappingTreads.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    this.focusWaste += event.waste;
+    this.possibleGain += FOCUS_PER_PROC;
+  }
+
+  item() {
+    return {
+      item: ITEMS.NESINGWARYS_TRAPPING_TREADS,
+      result: (
+        <dfn data-tip={`You procced Nesingwary's Trapping Treads ${this.possibleGain / FOCUS_PER_PROC} times.`}>
+          <span>You gained {this.focusGain} out of {this.possibleGain} possible focus.</span>
```
Comment (kfinch) [inline src/Parser/Hunter/Survival/Modules/Items/NesingwarysTrappingTreads.js]: Would be better to show a fight length independant number imo, like "Focus gained per minute". Also relegate the overcapped amount to the tooltip IMO.

Also, `<Wrapper>` pls

File: src/Parser/Hunter/Survival/Modules/Items/UnseenPredatorsCloak.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  get uptimePercentage() {
+    return this.enemies.getBuffUptime(SPELLS.UNSEEN_PREDATORS_CLOAK_BUFF.id) / this.owner.fightDuration;
+  }
+
+  item() {
+    return {
+      item: ITEMS.UNSEEN_PREDATORS_CLOAK,
+      result: <Wrapper>You had {formatPercentage(this.uptimePercentage)}% uptime on <SpellLink id={SPELLS.UNSEEN_PREDATORS_CLOAK_BUFF.id} />.</Wrapper>,
```
Comment (kfinch) [inline src/Parser/Hunter/Survival/Modules/Items/UnseenPredatorsCloak.js]: Tooltip-level wordiness, could just be `{formatPercentage(this.uptimePercentage)}% uptime`, the item tooltip already present by icon mouseover.

File: src/Parser/Hunter/Survival/Modules/Items/HelbrineRopeOfTheMistMarauder.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  item() {
+    return {
+      item: ITEMS.HELBRINE_ROPE_OF_THE_MIST_MARAUDER,
+      result: (
+        <dfn data-tip={`You applied the Mark of Helbrine debuff ${this.applications} times. </br> The reason this shows as "up to X dmg" is because the tooltip has no information on the potency of the applied Mark of Helbrine. To maximize the potential of this legendary, you want to stand as far away as possible and cast Harpoon when engaging the first time with mobs.`}>
+          <Wrapper>
+            You had {formatPercentage(this.uptimePercentage)}% uptime on <SpellLink id={SPELLS.MARK_OF_HELBRINE.id} />. <br />
+            It contributed with up to <ItemDamageDone amount={this.bonusDamage} />
```
Comment (kfinch) [inline src/Parser/Hunter/Survival/Modules/Items/HelbrineRopeOfTheMistMarauder.js]: Save the wordiness for tooltip, top level display could just be 
```
<Wrapper>
  {formatPercentage(this.uptimePercentage)}% uptime<br />
  Up to <ItemDamageDone amount={this.bonusDamage} /> *
</Wrapper>
```

File: src/Parser/Hunter/Survival/Modules/Items/NesingwarysTrappingTreads.js (hunk 1 above)
Comment (Pewtro) [reply src/Parser/Hunter/Survival/Modules/Items/NesingwarysTrappingTreads.js]: I thought we didn't use `<Wrapper>` on things like this 🤔

Comment (MartijnHols) [reply src/Parser/Hunter/Survival/Modules/Items/NesingwarysTrappingTreads.js]: you don't use any extra wrapper for this, dfn already serves as one

**Pull request merged**